libcloudforensics/providers/gcp/internal/compute.py

```diff
@@ -401,9 +401,17 @@ def CreateDiskFromSnapshot(
     }
     try:
       gce_disks_client = self.GceApi().disks()
-      request = gce_disks_client.insert(
-          project=self.project_id, zone=self.default_zone, body=body)
-      response = request.execute()
+      while True:
+        try:
+          request = gce_disks_client.insert(
+              project=self.project_id, zone=self.default_zone, body=body)
+          response = request.execute()
+          break
+        except HttpError as exception:
+          if 'is not ready' not in str(exception):
```

Comment:
Rather than retrying until it's ready is there somewhere that we can use [BlockOperation](https://github.com/google/cloud-forensics-utils/blob/985c723b5684b6c3caedb9d16ef67bf6abf34562/libcloudforensics/providers/gcp/internal/common.py#L240) to ensure this function isn't called until the resources are ready?